games-simulation/openrct2/openrct2-0.2.4.ebuild

```diff
@@ -0,0 +1,182 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils xdg-utils
+
+MY_PN="OpenRCT2"
+MY_PN_OBJ="objects"
+MY_PN_RPL="replays"
+MY_PN_TS="title-sequences"
+MY_PV_OBJ="1.0.12"
+MY_PV_RPL="0.0.4"
+MY_PV_TS="0.1.2c"
+
+DESCRIPTION="An open source re-implementation of Chris Sawyer's RollerCoaster Tycoon 2"
+HOMEPAGE="https://openrct2.org/"
+SRC_URI="
+	https://github.com/${MY_PN}/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/${MY_PN}/${MY_PN_OBJ}/releases/download/v${MY_PV_OBJ}/${MY_PN_OBJ}.zip -> ${PN}-${MY_PN_OBJ}-${MY_PV_OBJ}.zip
+	https://github.com/${MY_PN}/${MY_PN_TS}/releases/download/v${MY_PV_TS}/${MY_PN_TS}.zip -> ${PN}-${MY_PN_TS}-${MY_PV_TS}.zip
+	test? ( https://github.com/${MY_PN}/${MY_PN_RPL}/releases/download/v${MY_PV_RPL}/${MY_PN_RPL}.zip -> ${PN}-${MY_PN_RPL}-${MY_PV_RPL}.zip )
+"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dedicated libressl +lightfx +multiplayer +opengl test +truetype twitch"
+REQUIRED_USE="dedicated? ( !lightfx multiplayer !opengl !truetype !twitch )"
```

Comment:
There seems to be something wrong with this logic but I can't quite put my finger on it yet, I'll need some extra time to figure it out.

For example, according to this logic `dedicated` requires `multiplayer`. I disabled this line and built the package with `USE='dedicated -lightfx -multiplayer -opengl -truetype -twitch'`. The resulting `openrct2-cli` happily opens my save game of Diamond Heights from RCT1 Deluxe and lets me edit it (although it seems to corrupt the save game file).

I wonder if there is any behavioral difference between USE="dedicated lightfx" and USE="dedicated -lightfx" for example, if you have a client connecting to your headless server. It seems like there is none. It will build libopenrct2 with lightfx but who will consume it if GUI is disabled?

Reply:
It is correct, that even if you are building a dedicated server only, the build system allows you to enable `lightfx`. I asked upstream and they said, that it won't changed anything, as it is not used at all for the dedicated server. So I blocked that on purpose, so a user won't have false expectations. Same goes for the other use flags. These are all features, which are not used at all, despite some of them being allowed to be enabled by the build system when building.

About the `multiplayer` use flag. Did you ran a test in your network or locally only? I observed, that without enabled `multiplayer`, I am only able to connect locally, but not from my network to the server. Did you tested that and it worked for you? That's the reason, why I am forcing the use flag.

Reply:
> About the `multiplayer` use flag. Did you ran a test in your network or locally only? I observed, that without enabled `multiplayer`, I am only able to connect locally, but not from my network to the server. Did you tested that and it worked for you? That's the reason, why I am forcing the use flag, since a local only dedicated server does not make sense to me. Note: The daemon for the server, despite of `multiplayer` enabled or disabled, will always successfully start.

I didn't test the server capabilities actually. I used the cli to manipulate a save game only. My major annoyance with this is that I can't just `USE=dedicated emerge openrct2` because it'll force me disable a bunch of USE flags (due to REQUIRED_IUSE) and when you think about it the flags I am being forced to disable are not even relevant to the build as they cannot be possibly enabled anyway. In other words, the dependency established in REQUIRED_IUSE is sort of fake: when you force opengl to be disabled, nothing in the code will change as it's not including the GUI code anyway. In effect `USE='dedicated opengl'` and `USE='dedicated -opengl'` would result in the same binary being built. Let me ask you what would the downside of dropping REQUIRED_IUSE altogether be and we can work on how to deal with this after that.

Reply:
After more digging into it, in order to make it more user friendly, I will drop `multiplayer` and enable unconditionally. Since it depends only on `libssl`, that should not be a problem.

As for the rest of use flags, it's okay to drop them from `REQUIRED_IUSE`. So I have removed that now entirely and shifted some dependencies.

My 'fake' REQUIRED_IUSE was only meant to be not producing false expectations for an user. So If someone enables `dedicated` and also `twitch`, it won't have twitch functionalty. That's all.
Technically, nothing speaks against it, as with enabled dedicated, no GUI is build.